test/drenv/envfile.py

```diff
@@ -83,20 +88,24 @@ def _validate_profile(profile):
     profile.setdefault("workers", [])
 
     for i, worker in enumerate(profile["workers"]):
-        _validate_worker(worker, profile, i)
+        _validate_worker(worker, profile, addons_dir, i)
 
 
-def _validate_worker(worker, env, index):
+def _validate_worker(worker, env, addons_dir, index):
     worker["name"] = f'{env["name"]}/{worker.get("name", index)}'
     worker.setdefault("addons", [])
 
     for addon in worker["addons"]:
-        _validate_addon(addon, env, args=[env["name"]])
+        _validate_addon(addon, env, addons_dir, args=[env["name"]])
 
 
-def _validate_addon(addon, env, args=()):
+def _validate_addon(addon, env, addons_dir, args=()):
```

Comment:
Having to pass addons_dir to so many layers is not great but we can improve this later.